### PR TITLE
Add error for taking table with no edge metadata

### LINKED ISSUE #2207: Check behaviour is tested for the combination of `TSK_NO_EDGE_METADATA` and `TSK_TAKE_OWNERSHIP`
We have a note about this at `trees.c:417` but I don't think a test was added when `TSK_TAKE_TABLES` was added.

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #2207 
Added an error for this case.

### CODE REVIEW

File: c/tskit/trees.c [hunk 1]
```diff
@@ -407,6 +407,10 @@ tsk_treeseq_init(
     tsk_memset(self, 0, sizeof(*self));
     if (options & TSK_TAKE_OWNERSHIP) {
         self->tables = tables;
+        if (tables->edges.options & TSK_NO_EDGE_METADATA) {
```
Comment (jeromekelleher) [inline c/tskit/trees.c:410]: Shouldn't the flag be TSK_TABLE_NO_METADATA though? I thought that was the point of that flag - any table that has no metadata would use it. the TSK_TC_NO_EDGE_METADATA was a way to send that flag to edge_table_init.

Maybe we should wait until #2224 is in before tackling this.

**Review by jeromekelleher — COMMENTED** [review]
Good catch, but I think the implementation is uncovering another bug?

File: c/tskit/trees.c (hunk 1 above)
Comment (benjeffery) [reply c/tskit/trees.c]: Ah, they have the same value as I haven't rebased, so if I had rebased my mistake would have been revealed. 

I've rebased and fixed this.

**Comment by benjeffery** [comment]
@jeromekelleher I think this is ready to now.

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**